pkg/resources/reconciler.go

```diff
@@ -219,6 +218,24 @@ func (r *Reconciler) ReconcileSubscription(ctx context.Context, target marketpla
 			return integreatlyv1alpha1.PhaseFailed, fmt.Errorf("error approving installplan for %v: %w", target.Pkg, err)
 		}
 
+		// Workaround to re-install product operator if install plan fails due to https://bugzilla.redhat.com/show_bug.cgi?id=1923111
+		if ip.Status.Phase == operatorsv1alpha1.InstallPlanPhaseFailed {
+			if sub.Status.InstalledCSV != "" {
+				log.Warningf("Deleting csv for re-install due to failed install plan", l.Fields{"ns": target.Namespace, "install plan": target.Pkg, "csv": sub.Status.InstalledCSV})
+				csv := &operatorsv1alpha1.ClusterServiceVersion{
+					ObjectMeta: metav1.ObjectMeta{Namespace: target.Namespace, Name: sub.Status.InstalledCSV},
+				}
+				if err := client.Delete(ctx, csv); err != nil && k8serr.IsNotFound(err) {
```

Comment:
curious why we have to remove the CSV?

Reply:
Before adding this change to remove the csv also, if we remove the subscription without removing the installed csv, a new subscription would be created due but a install plan wont be created since the csv is already/still installed in the namespace preventing re-install from continuing

When the csv is deleted, the operator pod is deleted. Deleting the subscription only cleans up the install plans it seems. 
Uninstalling the operator via the ui looks to perform both these tasks of deleting subscription and csv in the namespace :thinking: